Formula/sonarqube.rb

```diff
@@ -25,34 +26,62 @@ class Sonarqube < Formula
   conflicts_with "sonarqube-lts", because: "both install the same binaries"
 
   def install
-    # Use Java Service Wrapper 3.5.46 which is Apple Silicon compatible
-    # Java Service Wrapper doesn't support the  wrapper binary to be symlinked, so it's copied
+    # Delete older packaged java-service-wrapper
+    wrapper_version = "3.2.3"
+    buildpath.glob("bin/*-64").map(&:rmtree)
+    (buildpath/"lib/jsw/wrapper-#{wrapper_version}.jar").unlink
+
+    platform = OS.mac? ? "macosx-universal-64" : "linux-x86-64"
+    platform_bin = buildpath/"bin"/platform
+    (platform_bin/"lib").mkpath
+
+    # Link newer java-service-wrapper formula which is Apple Silicon compatible
     jsw_libexec = Formula["java-service-wrapper"].opt_libexec
-    ln_s jsw_libexec/"lib/wrapper.jar", "#{buildpath}/lib/jsw/wrapper-3.5.46.jar"
-    ln_s jsw_libexec/"lib/libwrapper.dylib", "#{buildpath}/bin/macosx-universal-64/lib/"
-    cp jsw_libexec/"bin/wrapper", "#{buildpath}/bin/macosx-universal-64/"
-    cp jsw_libexec/"scripts/App.sh.in", "#{buildpath}/bin/macosx-universal-64/sonar.sh"
-    sonar_sh_file = "bin/macosx-universal-64/sonar.sh"
-    inreplace sonar_sh_file, "@app.name@", "SonarQube"
-    inreplace sonar_sh_file, "@app.long.name@", "SonarQube"
-    inreplace sonar_sh_file, "../conf/wrapper.conf", "../../conf/wrapper.conf"
-    inreplace "conf/wrapper.conf", "wrapper-3.2.3.jar", "wrapper-3.5.46.jar"
-    rm "lib/jsw/wrapper-3.2.3.jar"
-    rm "bin/macosx-universal-64/lib/libwrapper.jnilib"
+    ln_s jsw_libexec/"lib/wrapper.jar", buildpath/"lib/jsw/wrapper.jar"
+    ln_s jsw_libexec/"lib"/shared_library("libwrapper"), platform_bin/"lib/"
+    ln_s jsw_libexec/"bin/wrapper", platform_bin/"wrapper"
 
-    # Delete native bin directories for other systems
-    remove, keep = if OS.mac?
-      ["linux", "macosx-universal"]
-    else
-      ["macosx", "linux-x86"]
+    # Create new sonar.sh script and update conf files
+    cp jsw_libexec/"scripts/App.sh.in", platform_bin/"sonar.sh"
+    chmod 0755, platform_bin/"sonar.sh"
+    inreplace platform_bin/"sonar.sh" do |s|
+      s.gsub! "@app.name@", "SonarQube"
+      s.gsub! "@app.long.name@", "SonarQube"
+      s.gsub! "../conf/wrapper.conf", "../../conf/wrapper.conf"
+      # Write PID file outside of installation directory
+      s.sub!(/^PIDDIR="\."$/, "PIDDIR=#{var}/run")
+    end
+    inreplace "conf/wrapper.conf" do |s|
+      s.gsub! "wrapper-#{wrapper_version}.jar", "wrapper.jar"
+      # Set wrapper.working.dir to allow using symlinked wrapper
+      s.sub!(/^wrapper\.java\.command=.*/, "\\0\nwrapper.working.dir=#{libexec}/bin/#{platform}")
+      # Write log files outside of installation directory
+      s.sub! %r{^wrapper\.logfile=.*/([^/]+\.log)$}, "wrapper.logfile=#{var}/sonarqube/logs/\\1"
```

Comment:
Before merge, I will probably update regex to avoid greedy capture of whitespace/newlines (`[^/]+`)

Maybe `[^/\s]+`.

Reply:
```suggestion
      s.sub! %r{^wrapper\.logfile=.*/([^/\s]+\.log)$}, "wrapper.logfile=#{var}/sonarqube/logs/\\1"
```